projects/mci/test/test_mci_fan_control.c

```diff
@@ -0,0 +1,175 @@
+#include "mci_fan_control.h"
+
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "ms_test_helpers.h"
+
+// Initialize fan control with no callbacks.
+static void prv_init_fan(void) {
+  MciFanControlSettings settings = {
+    .fault_cb = NULL,
+    .fault_context = NULL,
+  };
+  TEST_ASSERT_OK(mci_fan_control_init(&settings));
+}
+
+// Used to make sure the fault callback is working as expected
+static volatile uint16_t s_times_cb_called;

```

Comment:
Though these static variables are automatically initialized to 0, I think we should still initialize them ourselves.
Also, does this need to be volatile? It seems like this variable is only used within this test and not modified elsewhere (ie we aren't reading from hardware registers where something else could modify the contents stored in this variable)

Reply:
I think the volatile got left in when I was trying to debug a weird testing bug I introduced a while back, whoops, good catch, thanks